identity-account-storage/src/types/key_location.rs

```diff
@@ -5,61 +5,163 @@ use core::fmt::Debug;
 use core::fmt::Display;
 use core::fmt::Formatter;
 use core::fmt::Result;
-use identity_core::common::Fragment;
+use identity_core::crypto::KeyType;
+use identity_did::verification::MethodData;
 use identity_did::verification::MethodType;
-use serde::Deserialize;
-use serde::Serialize;
-
-use crate::types::Generation;
+use identity_iota_core::document::IotaVerificationMethod;
+use seahash::SeaHasher;
+use std::hash::Hash;
+use std::hash::Hasher;
 
 /// The storage location of a verification method key.
-#[derive(Clone, Hash, PartialEq, Eq, Deserialize, Serialize)]
+///
+/// A key is uniquely identified by the fragment and a hash of its public key.
+/// Importantly, the fragment alone is insufficient to represent the storage location.
+/// For example, when rotating a key, there will be two keys in storage for the
+/// same identity with the same fragment. The `key_hash` disambiguates the keys in
+/// situations like these.
+///
+/// The string representation of that location can be obtained via `canonical_repr`.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
 pub struct KeyLocation {
-  method: MethodType,
-  fragment: Fragment,
-  generation: Generation,
+  /// The [`KeyType`] of the key.
+  pub key_type: KeyType,
+  /// The fragment of the key.
+  fragment: String,
+  /// The hash of the public key.
+  #[serde(with = "key_hash_serialization")]
+  pub(in crate::types::key_location) key_hash: u64,
```

Comment:
Shouldn't we just use a `String` to store the hash if it's giving problems with serialization, or do we want to enforce it's always a number?